tests/Doctrine/Tests/ORM/Mapping/XmlMappingDriverTest.php

```diff
@@ -163,32 +163,88 @@ public function testInvalidMappingFileException(): void
      * @dataProvider dataValidSchema
      * @group DDC-2429
      */
-    public function testValidateXmlSchema(string $xmlMappingFile): void
-    {
-        $xsdSchemaFile = __DIR__ . '/../../../../../doctrine-mapping.xsd';
-        $dom           = new DOMDocument();
-
-        $dom->load($xmlMappingFile);
-
-        self::assertTrue($dom->schemaValidate($xsdSchemaFile));
+    public function testValidateXmlSchema(
+        string $class,
+        string $tableName,
+        array $fieldNames,
+        array $associationNames
+    ): void {
+        $metadata = $this->createClassMetadata($class);
+
+        $this->assertInstanceOf(ClassMetadata::class, $metadata);
+        $this->assertEquals($metadata->getTableName(), $tableName);
+        $this->assertEquals($metadata->getFieldNames(), $fieldNames);
+        $this->assertEquals($metadata->getAssociationNames(), $associationNames);
     }
 
     /**
-     * @psalm-return list<array{string}>
+     * @psalm-return []array{0: class-string, 1: string, 2: list<string>, 3: list<string>}
      */
     public static function dataValidSchema(): array
     {
-        $list    = glob(__DIR__ . '/xml/*.xml');
-        $invalid = ['Doctrine.Tests.Models.DDC889.DDC889Class.dcm'];
-        assert(is_array($list));
+        return [
+            [
+                User::class,
+                'cms_users',
+                ['name', 'email', 'version', 'id'],
+                ['address', 'phonenumbers', 'groups'],
+            ],
+            [
+                DDC889Entity::class,
+                'DDC889Entity',
+                [],
+                [],
+            ],
+            [
+                DDC889SuperClass::class,
+                'DDC889SuperClass',
+                ['name'],
+                [],
+            ],
+        ];
+    }
 
-        $list = array_filter($list, static function (string $item) use ($invalid): bool {
-            return ! in_array(pathinfo($item, PATHINFO_FILENAME), $invalid, true);
-        });
+    /**
+     * @param array{0: class-string, 1: array<string, int>} $expectedExceptionOccurrences
+     *
+     * @dataProvider dataInvalidSchema
+     */
+    public function testValidateIncorrectXmlSchema(string $class, array $expectedExceptionOccurrences): void
+    {
+        try {
+            $this->createClassMetadata($class);
+        } catch (Throwable $throwable) {
+            $this->assertInstanceOf(MappingException::class, $throwable);
+
```

Comment:
```suggestion
        } catch (MappingException $throwable) {
```

Additionally, we need to make sure the test fails if no exception is raised at all.